lib/ansible/modules/cloud/vmware/vmware_guest.py

```diff
@@ -2398,6 +2398,9 @@ def deploy_vm(self):
         for nw in self.params['networks']:
             for key in nw:
                 # We don't need customizations for these keys
+                if key == 'type' and nw['type'] == 'dhcp':
```

Comment:
Add this check inside if key not in ('device_type', 'mac', 'name', 'vlan', 'type', 'start_connected'): condition. Dont make it separate if block

Reply:
Sorry, I misread the logic. Look good to me.